### PR TITLE
Attempt to implement Graphql ISO Date

### PR DESCRIPTION
**Pull request by natac13**
Ref #53
**Not ready for merge**
So I have not fully succeeded.
However I wanted to created the PR just incase you were interested and could lend a hand to help me finish.

Also I have not figured out how to make an Array of Dates yet. Although I am starting to undertand maybe what has to be done.

[L:90](https://github.com/arackaf/mongo-graphql-starter/compare/master...natac13:natac13/graphql-iso-date?expand=1#diff-60e96d7c14a1711e72251f1cc0dcbba0R90) is where I cannot figure out how to get the comparator operators working.

### CODE REVIEW

**Comment by arackaf** [comment]
Hey thanks for the work! I’d definitely be up for helping finish this, but no promises on how soon. On cursory read, I think this line

return type !== JSONType || type !== ISODateType || type !== ISODateTimeType || type !== ISOTimeType

should use && instead of ||, shouldn’t it?

I’ll take a closer look when I’m back home in a few days.

**Comment by natac13** [comment]
Hi Adam,
> should use && instead of ||, shouldn’t it?

Yep! I seem to get that mixed up quite a bit... :disappointed: 

> Hey thanks for the work!

No problem. I just wanted to show my appreciation for the work that you have put into this project!

I found this project from a Google search to issue #2. I first forked and made an attempt to create a PR for that issue. Especially with how well you laid out the [requirements](https://github.com/arackaf/mongo-graphql-starter/issues/2#issuecomment-411829831). However I was running into some issues with finding a way to match the mongoose [types](https://mongoosejs.com/docs/schematypes.html) to the [types](https://github.com/arackaf/mongo-graphql-starter/blob/master/src/dataTypes.js#L3-L13) your package is expecting. For example the `Int` & `Float`. I would assume that float is not the Deciaml128 type in MongoDB. The issue is that mongoose only has Number type. Therefore I thought of adding a `meta` field to the mongoose model. But that did not seem right.....

However, I really like the fact of how much graphql code your project would generate. And the fact that you have the mongo project generated from the `ast` is **amazing**. I have been struggling with this aspect and have been just hard coding projects, which is horrible!  Therefore I thought to and re-write my endpoint without mongoose while using your library. The issue came when I tried to model an array of Dates. Which lead me to this issue.

And since I was unable to directly utilize this project right away I wanted to help out to make it work. 
At the moment I have used your project to generate as close to possible my database model shapes. this allowed me to use the `generateSchema.gql` file and *upgrade* my schema definitions. While slowly understanding the functions in [projectUtilities.js](https://github.com/arackaf/mongo-graphql-starter/blob/master/src/projectUtilities.js) and [queryUtilities.js](https://github.com/arackaf/mongo-graphql-starter/blob/master/src/queryUtilities.js). I am re-writing the logic to work with my mongoose graphql endpoint. This I am hoping my lead to me being able to figure out the PR to #2 

Sorry for the long reply. 
In short thanks you for this project.

I do apologize as well, as the issue opened was about an DateArrayType and I did a PR for a new Date type. I just thought it would be a good first step into the code base to figure things out and then use the new ISODate with the DateArrayType if I could figure that out.

**Comment by arackaf** [comment]
Hey - cool seeing someone so interested in something I made :)

I'm traveling now - hope to tackle this, and these issues tomorrow, or over the weekend.

**Comment by arackaf** [comment]
Hey so ... are you still needing help with this? Can you lay out exactly where with this PR you're stuck and struggling?

**Comment by natac13** [comment]
Hi Adam, Sorry I have been pretty busy lately. I have a project that launches tuesday and I have somethings I should fix before then. 

I can come back to this once I am done. Feel free to close this PR if you would like and I can open another once I have started working on this again. Then I can let you know if I am still stuck.

Thanks

**Comment by arackaf** [comment]
Oh cool - no pressure in the least, of course, and I have no reason to need to close this. Come back whenever and let me know if there's anything I can do to help.

**Comment by arackaf** [comment]
Yo. I just re-wrote a bunch of history in my repo, to fix some of my old email addresses. As a result, this PR is now off by hundreds of commits. It's not a problem in the least. *IF* you want to continue work on this, just please do a pull of my upstream, and then **rebase** this branch onto my master (and then open a new PR). If you need any help at all with that let me know and I'd be happy to.

(note that it seems the branch only has one commit - if that's right just grab my master and cherry pick that lone commit onto a new branch of yours.)
